src/router.tsx

```diff
@@ -0,0 +1,175 @@
+import {
+  Link,
+  HashRouter,
+  BrowserRouter,
+  Routes,
+  Route,
+  Outlet,
+} from 'react-router-dom';
+import App from './containers/application/application';
+import SearchResults from './containers/Search/SearchResults';
+import Wallet from './containers/Wallet/Wallet';
+import Home from './containers/home/home';
+import Governance from './containers/governance/governance';
+import ProposalsDetail from './containers/governance/proposalsDetail';
+import Validators from './containers/Validators/Validators';
+import Story from './containers/story/story';
+import TxsDetails from './containers/txs/txsDetails';
+import AccountDetails from './containers/account';
+import ValidatorsDetails from './containers/validator';
+import Ipfs from './containers/ipfs/ipfs';
+import BlockDetails from './containers/blok/blockDetails';
+import Txs from './containers/txs';
+import Block from './containers/blok';
+import ParamNetwork from './containers/parameters';
+import TrollBoxx from './containers/trollBox';
+import ForceGraph from './containers/forceGraph/forceGraph';
+import ForceQuitter from './containers/forceGraph/forceQuitter';
+import TestKeplr from './containers/testKeplre';
+import Mint from './containers/mint';
+import RoutedEnergy from './containers/energy';
+import Market from './containers/market';
+import Oracle from './containers/oracle';
+import Objects from './containers/Objects';
+import Taverna from './containers/taverna';
+import Teleport from './containers/teleport';
+import Nebula from './containers/nebula';
+import Movie from './containers/movie';
+import PortalCitizenship from './containers/portal';
+import PortalGift from './containers/portal/gift';
+import Release from './containers/portal/release';
+import Temple from './containers/temple';
+import IpfsSettings from './containers/ipfsSettings';
+import Ibc from './containers/ibc';
+import {
+  Codes,
+  CodePage,
+  ContractPage,
+  DashboardPage,
+} from './containers/wasm';
+import Help from './containers/help';
+import Assets from './containers/assets';
+import MainPartal from './containers/portal/mainPortal';
+import {
+  ListNetwork,
+  CustomNetwork,
+  DetailsNetwork,
+} from './containers/network';
+
+import Sigma from './containers/sigma';
+
+import { routes } from './routes';
+import Warp from './containers/warp/Warp';
+
+type WrappedRouterProps = {
+  children: React.ReactNode;
+};
+
+function WrappedRouter({ children }: WrappedRouterProps) {
+  return process.env.IPFS_DEPLOY ? (
+    <HashRouter>{children}</HashRouter>
+  ) : (
+    <BrowserRouter>{children}</BrowserRouter>
+  );
+}
+
+function PageNotExist() {
+  return (
+    <div>
+      page not exists
+      <br />
+      <Link to={routes.home.path}>Home</Link>
+    </div>
+  );
+}
+
+function MainLayout() {
+  return (
+    <>
+      <App />
+      <Outlet />
+    </>
+  );
+}
+
+function AppRouter() {
+  return (
+    <WrappedRouter>
+      <Routes>
+        <Route path="/" element={<MainLayout />}>
```

Comment:
why you removed `routes` config? it is better than strings

Reply:
we need to choose config or just strings, mixing no sense. You can refactor that later)